pkg/util/struct.go

```diff
@@ -0,0 +1,48 @@
+package util
+
+import "reflect"
+
+// VisitAllStringFields visits all string fields in the given pointer to struct
+// and allow modifying them with visitor function
+func VisitAllStringFields(obj interface{}, visitor func(string) string) {
```

Comment:
i tried to make this specific for 'string' values. If I make this more generic, you will have to protect yourself using CanSet() etc and it will make the visitor() func more complex.